frontend/src/metabase/dashboard/components/ClickBehaviorSidebar/LinkOptions/LinkedEntityPicker.tsx

```diff
@@ -157,6 +178,61 @@ function LinkedEntityPicker({
     });
   }, [clickBehavior, updateSettings]);
 
+  const { data: dashboard } = useDashboardQuery({
+    enabled: isDashboard,
+    id: targetId,
+  });
+  const dashboardTabs = dashboard?.tabs ?? NO_DASHBOARD_TABS;
+  const defaultDashboardTabId: number | undefined = dashboardTabs[0]?.id;
+  const dashboardTabId = isDashboard
+    ? clickBehavior.tabId ?? defaultDashboardTabId
+    : undefined;
+  const dashboardTabExists = dashboardTabs.some(
+    tab => tab.id === dashboardTabId,
+  );
+  const dashboardTabIdValue =
+    typeof dashboardTabId === "undefined" ? undefined : String(dashboardTabId);
+
+  const handleDashboardTabChange = (value: string) => {
+    if (!isDashboard) {
+      throw new Error("This should never happen");
+    }
+
+    updateSettings({ ...clickBehavior, tabId: Number(value) });
+  };
+
+  useEffect(
+    function migrateUndefinedDashboardTabId() {
+      if (
+        isDashboard &&
+        typeof clickBehavior.tabId === "undefined" &&
+        typeof defaultDashboardTabId !== "undefined"
+      ) {
+        updateSettings({ ...clickBehavior, tabId: defaultDashboardTabId });
+      }
+    },
+    [clickBehavior, defaultDashboardTabId, isDashboard, updateSettings],
+  );
+
+  useEffect(
+    function migrateDeletedTab() {
```

Comment:
I think that's incorrect. If will show the default tab when an invalid one was previously selected, but without updating the click behavior the "migration" won't take affect.

Reply:
Even without it taking effect, the to auto-url-cleanup in dashboard page will compensate for this, and it behaves the same (i.e. redirects to the default tab).

Perhaps the `migrate` prefix is a bit misleading?